eng/scripts/TypeSpec-Generate-Sdk.ps1

```diff
@@ -0,0 +1,35 @@
+[CmdletBinding()]
+param (
+  [Parameter(Position = 0)]
+  [ValidateNotNullOrEmpty()]
+  [string] $SdkRepoRootDirectory,
+  [Parameter(Position = 1)]
+  [string] $TypeSpecProjectDirectory, # A directory of `tspconfig.yaml` or a remoteUrl of `tspconfig.yaml`
+  [Parameter(Position = 2)]
+  [string] $CommitHash,
```

Comment:
Can we also create reasonable defaults for these base on calling git commands to retrieve them?

Reply:
Or if that is handled by the process script that is fine as well.

Reply:
Normally in local scenario, user would like to build sdk upon the typespecs within local folder rather than fetch spec from a remote commit. And the local typespec changes aren't commited yet. So, leave commitHash and repoUrl empty is this case.

The other hand only when tsp-location.yaml doesn't exist then it would need to compute the commitHash and repoUrl to create tsp-location.yaml.  With local spec repo, yes it can calculate the commitHash. I would update the `TyepSpec-Project-Process.ps1` to add a input parameter of `generateFromLocalTypeSpec`. If the flag is true, then script would get the local spec repo commit and use that to create tsp-location.yaml but always generate sdk based on the local typespecs. What do you think of this idea? @weshaggard

Reply:
Hopefully we would need to add the new "generatefromlocaltypespec" parameter to the process script but instead compute them and pass the values if needed.